src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -938,26 +948,29 @@ StoragePtr InterpreterSystemQuery::tryRestartReplica(const StorageID & replica,
     auto constraints = InterpreterCreateQuery::getConstraintsDescription(create.columns_list->constraints, columns, system_context);
     auto data_path = database->getTableDataPath(create);
 
-    table = StorageFactory::instance().get(create,
+    auto new_table = StorageFactory::instance().get(create,
         data_path,
         system_context,
         system_context->getGlobalContext(),
         columns,
         constraints,
         LoadingStrictnessLevel::ATTACH);
 
-    database->attachTable(system_context, replica.table_name, table, data_path);
+    database->attachTable(system_context, replica.table_name, new_table, data_path);
+    if (new_table->getStorageID().uuid != replica_table_id.uuid)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Tables UUID does not match after RESTART REPLICA (old: {}, new: {})", replica_table_id.uuid, new_table->getStorageID().uuid);
+    if (new_table->is_being_restarted)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "RESTORed flag is set for table {}", new_table->getStorageID().getNameForLogs());
```

Comment:
this message looks a bit weird

Reply:
The problem is that without https://github.com/ClickHouse/ClickHouse/pull/76308 you may set `is_being_restored` to one table and restore it to another, so this is just a sanity check, which, should never fail, since newly created table have this flag set to `false` by default.

Reply:
This is actually the initial problem that I was looking at - https://github.com/ClickHouse/ClickHouse/pull/76308

Reply:
Actually it should not be possible, I removed it.